packages/shared/src/util/features.ts

```diff
@@ -136,6 +146,104 @@ export function validateFeatureValue(
+const isForceRule = (rule: FeatureRule): rule is ForceRule =>
+  rule.type === "force";
+const isExperimentRule = (rule: FeatureRule): rule is ExperimentRefRule =>
+  rule.type === "experiment" || rule.type === "experiment-ref";
+
+const areRulesOneSided = (
+  rules: FeatureRule[], // can assume all rules are enabled
+  linkedExperiments: ExperimentInterfaceStringDates[]
+) => {
+  const rolloutRules = rules.filter(isRolloutRule);
+  const forceRules = rules.filter(isForceRule);
+  const expRules = rules.filter(isExperimentRule);
+
+  const rolloutRulesOnesided =
+    !rolloutRules.length ||
+    rolloutRules.every((r) => r.coverage === 1 && !r.condition);
+
+  const forceRulesOnesided =
+    !forceRules.length || forceRules.every((r) => !r.condition);
+
+  const expRulesOnesided =
+    !expRules.length ||
+    expRules.every((r) => {
+      const linkedExp = linkedExperiments.find((e) => e.id === r.experimentId);
+      return linkedExp?.status === "draft" || linkedExp?.status === "stopped";
+    });
+
+  return rolloutRulesOnesided && forceRulesOnesided && expRulesOnesided;
+};
+
+export function isFeatureStale(
+  feature: FeatureInterface,
+  linkedExperiments: ExperimentInterfaceStringDates[] | undefined = []
+): { stale: boolean; reason?: StaleFeatureReason } {
+  if (feature.neverStale) return { stale: false };
+
+  if (feature.linkedExperiments?.length !== linkedExperiments.length) {
+    // eslint-disable-next-line no-console
+    console.error("isFeatureStale: linkedExperiments length mismatch");
+    return { stale: false, reason: "error" };
+  }
+
+  const linkedExperimentIds = linkedExperiments.map((e) => e.id);
+  if (
+    !linkedExperimentIds.every((id) => feature.linkedExperiments?.includes(id))
+  ) {
+    // eslint-disable-next-line no-console
+    console.error("isFeatureStale: linkedExperiments id mismatch");
+    return { stale: false, reason: "error" };
+  }
+
+  const twoWeeksAgo = subWeeks(new Date(), 2);
+  const dateUpdated = getValidDate(feature.dateUpdated);
+  const stale = dateUpdated < twoWeeksAgo;
+
+  if (!stale) return { stale };
+
+  // features with draft revisions are not stale
+  if (feature.hasDrafts) return { stale: false };
+
+  const envSettings = Object.values(feature.environmentSettings ?? {});
+
+  const enabledEnvs = envSettings.filter((e) => e.enabled);
+  const enabledRules = enabledEnvs
+    .map((e) => e.rules)
+    .flat()
+    .filter((r) => r.enabled);
+
+  if (enabledRules.length === 0) return { stale, reason: "no-rules" };
+
+  // features with only exp rules that are inactive are stale
+  if (
+    enabledRules.every(
+      (r) => r.type === "experiment" || r.type === "experiment-ref"
+    )
+  ) {
+    // TODO add test for experiments that are draft or stopped
+    const noExpsActive =
+      !!linkedExperiments.length &&
+      !linkedExperiments.some((e) => includeExperimentInPayload(e));
+    if (noExpsActive) return { stale, reason: "no-active-exps" };
+  }
```

Comment:
We should flip the logic here.  Easier to read and fixes a few edge cases.

```suggestion
  // If there's at least one active experiment, it's not stale
  if (linkedExperiments.som((e) => includeExperimentInPayload(e))) {
    return { stale: false }
  }
```

For a specific edge case that this fixes:
1. Experiment rule that is stopped with a temporary rollout enabled (included in experiment)
2. Force rule without any conditions

Should not be stale because of 1, but will return stale: true with a `rules-one-sided` reason

Reply:
Good catch, updated